lib/indico.js

```diff
@@ -6,35 +6,65 @@ var Promise = require('bluebird')
 
 Promise.promisifyAll(request);
 
-var services = [
+var textServices = [
   'political',
   'sentiment',
   'language',
+  'textTags',
+];
+
+var imageServices = [
   'facialFeatures',
   'fer',
   'imageFeatures',
-  'textTags'
 ];
 
+var otherServices = [
+  'predictText',
+  'predictImage'
+]
+
+var services = textServices.concat(imageServices).concat(otherServices)
+
 var public_api = 'http://apiv2.indico.io/'
 
 indico.apiKey = false;
 indico.cloud = false;
-function service(name, privateCloud, apiKey) {
+
+function service(name, privateCloud, apiKey, apis) {
+  /* given a set of parameters, returns the proper url for the REST api*/
   if (privateCloud) {
     var prefix = 'http://' + privateCloud + '.indico.domains/';
-    var url = prefix + name.toLowerCase();
+    var url = prefix + name;
   } else {
-    var url = public_api + name.toLowerCase();
+    var url = public_api + name;
   }
   url = url + "?key=" + apiKey;
+  if (apis !== undefined) {
+    url = url + "&apis=" + apis.join();
+  }
   return url
 }
 
-var api_request = function(api) {
-  var privateCloud, apiKey;
+function formatResults(results, nameMap) {
+  var formatted_results = {};
+  for (result in results) {
```

Comment:
Consider requiring lodash.forEach since for in loops are particularly slow in JavaScript.